frontend/src/hooks/useGenerateReportEffect.ts

```diff
@@ -1,56 +1,82 @@
-import { useState } from 'react'
+import { useRef, useState } from 'react'
 import { ERROR_MESSAGE_TIME_DURATION } from '@src/constants'
 import { reportClient } from '@src/clients/report/ReportClient'
 import { ReportRequestDTO } from '@src/clients/report/dto/request'
-import { reportMapper } from '@src/hooks/reportMapper/report'
-import { ReportDataWithThreeColumns, ReportDataWithTwoColumns } from '@src/hooks/reportMapper/reportUIDataStructure'
 import { UnknownException } from '@src/exceptions/UnkonwException'
 import { InternalServerException } from '@src/exceptions/InternalServerException'
+import { HttpStatusCode } from 'axios'
+import { reportMapper } from '@src/hooks/reportMapper/report'
+import { ReportResponse } from '@src/clients/report/dto/response'
 
 export interface useGenerateReportEffectInterface {
-  generateReport: (params: ReportRequestDTO) => Promise<
-    | {
-        velocityList?: ReportDataWithTwoColumns[]
-        cycleTimeList?: ReportDataWithTwoColumns[]
-        classification?: ReportDataWithThreeColumns[]
-        deploymentFrequencyList?: ReportDataWithThreeColumns[]
-        leadTimeForChangesList?: ReportDataWithThreeColumns[]
-        changeFailureRateList?: ReportDataWithThreeColumns[]
-      }
-    | undefined
-  >
+  startPollingReports: (params: ReportRequestDTO) => void
+  stopPollingReports: () => void
   isLoading: boolean
   isServerError: boolean
   errorMessage: string
+  reports: ReportResponse | undefined
 }
 
 export const useGenerateReportEffect = (): useGenerateReportEffectInterface => {
   const [isLoading, setIsLoading] = useState(false)
   const [isServerError, setIsServerError] = useState(false)
   const [errorMessage, setErrorMessage] = useState('')
+  const [reports, setReports] = useState<ReportResponse>()
+  const timerIdRef = useRef<NodeJS.Timer>()
 
-  const generateReport = async (params: ReportRequestDTO) => {
+  const startPollingReports = (params: ReportRequestDTO) => {
     setIsLoading(true)
-    try {
-      const res = await reportClient.report(params)
-      return reportMapper(res.response)
-    } catch (e) {
-      const err = e as Error
-      if (err instanceof InternalServerException || err instanceof UnknownException) {
-        setIsServerError(true)
-      } else {
-        setErrorMessage(`generate report: ${err.message}`)
-        setTimeout(() => {
-          setErrorMessage('')
-        }, ERROR_MESSAGE_TIME_DURATION)
-      }
-    } finally {
-      setIsLoading(false)
-    }
+    reportClient
```

Comment:
I need to handle the data in the hook, not in the ui component

Reply:
In this scenario, it's more tempting to use THEN, because it's background polling, and code-wise there's no need to render out the step of skipping a block of code for rendering, similar to a silent request in the background.
.then splits off from the original call and starts operating within its own scope, and will update at a time the original scope cannot predict. we can put semantics aside for a moment, it’s “more asynchronous,” because it leaves the old scope and branches off into a new one.